### ISSUE TITLE
Interplay with named graphs?

### ISSUE THREAD

**Issue by VladimirAlexiev** [w3c/rdf-star#25]
IMHO the spec should clarify the Interplay of rdf-star triples with named graphs.

I always thought that rdf-star triples are very similar to singleton graphs. Eg aren't these similar

```
:Alice :believes _:x.
_:x {:Bob :loves :Mary}

and

:Alice :believes << :Bob :loves :Mary>>
```

@gkellogg wrote in #16 "specify if a quoted triple used with a named graph is asserted to be in the same named graph, or in the default graph"

**Comment by pchampin** [comment]
You are right: this is one of the ways RDF* graphs can be encoded in standard RDF, as Olaf described in his papers. We should probably have a section on that in the spec.

@gkellog's remark in issue #16 is on a slightly topic, though: it is about the notion of RDF* *dataset* (which was not (I think) formally introduced in the RDF* papers) and how N-Quads and TriG can be extended to express them.

Note that encoding an RDF* dataset in an RDF dataset becomes non-trivial, because one would have to distinguish between "actual" named graphs, and "utility" named graph used only to encode embedded triples.

**Comment by hartig** [comment]
Like @pchampin says, an RDF* graph can be encoded syntactically into a standard RDF dataset that contains only single-triple named graphs. However, in the spec we are now also introducing the notion of an RDF* dataset. Perhaps this notion makes more clear what you are actually struggling with; namely, the distinction between an approach to capture annotations on the level of individual triples (which RDF* is for) and an approach to capture annotations on the level of whole graphs of triples (which named graphs are for). The notion of an RDF* dataset gives you both because it combines the notion of an RDF* graph with the notion of named graphs.

**Comment by ericprud** [comment]
Re-expressing to test my understanding of an RDF* dataset (which I'll call `Dataset*`) (ignoring literals for now):

`Dataset*` := `tuple[Graph*, map[Term->Graph*]]` # a default graph and map of named graphs
`Graph*` := `set[Triple*]`
`Triple*` := `tuple[Term*, IRI, Term*]`
`Term*`: `RdfTerm` | `Triple*` # add embedded triples to term
`RdfTerm` := `IRI` | `BNode`

When using conventional named graphs, we use predicates which imply a dereference, so in
```turtle
:Alice ng:believes _:x.
_:x {:Bob :loves :Mary}
```
, `ng:believes` asserts that the subject believes all of the statements appearing in the named graph identified by the object.

By contrast, an RDF* predicate is direct because the abstract syntax has been expanded to accommodate direct assertion, e.g.:
```turtle
:Alice rs:believes << :Bob :loves :Mary >>
```

In named graphs, it's trivial for a triple to refer (directly or indirectly) to the graph in which it appears.
```turtle
_:y { :ericP ng:expressed _:y }
```

Because there are no identifiers for embedded triples, it's impossible for one to reference itself.
```
:ericP rs:expressed << :ericP rs:expressed << :ericP rs:expressed rdf:nil >> >> .
```
(I though `rdf:nil` would be cute way to terminate that, but I don't intend any list semantics.)

**Comment by pchampin** [comment]
@ericprud your summary of RDF* datasets looks ok to me.

Indeed, RDF* triples can not refer to themselves, while named graphs can (modulo the fact that the graph "name" is not required by the semantics to denote the graph...).

**Comment by pchampin** [comment]
@VladimirAlexiev this issue has long been inactive. Do you object to closing it? It seems to me that the spec now clearly describes the interplay of of RDF-star triples with named graph, both at the abstract syntax and concrete syntax level (with N-Quads-star and TriG-star).
